graphReconstruct/graphReconstruct_test.go

```diff
@@ -1,46 +1,117 @@
+
+	chimpNode1 = &simpleGraph.Node{Id: 0, Name: "chimpNode1", Seq: dna.StringToBases("ACGT"), SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases("ACGT")), Prev: nil, Next: nil, Info: simpleGraph.Annotation{}}
+	chimpNode2 = &simpleGraph.Node{Id: 1, Name: "chimpNode2", Seq: dna.StringToBases("TTGG"), SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases("TTGG")), Prev: nil, Next: nil, Info: simpleGraph.Annotation{}}
+
+	chimpEdge1 = &simpleGraph.Edge{chimpNode2, 1.00}
+
+	chimpGraph = &simpleGraph.SimpleGraph{Nodes: []*simpleGraph.Node{chimpNode1, chimpNode2}}
+
+	gorillaNode1 = &simpleGraph.Node{Id: 0, Name: "gorillaNode1", Seq: dna.StringToBases("ACGT"), SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases("ACGT")), Prev: nil, Next: nil, Info: simpleGraph.Annotation{}}
+	gorillaNode2 = &simpleGraph.Node{Id: 1, Name: "gorillaNode2", Seq: dna.StringToBases("TTGG"), SeqTwoBit: dnaTwoBit.NewTwoBit(dna.StringToBases("TTGG")), Prev: nil, Next: nil, Info: simpleGraph.Annotation{}}
+
+	gorillaEdge1 = &simpleGraph.Edge{Dest: gorillaNode2, Prob: 1.00}
+
+	gorillaGraph = &simpleGraph.SimpleGraph{Nodes: []*simpleGraph.Node{gorillaNode1, gorillaNode2}}
+
+	nodeAlign0 = graphColumn{AlignId: 0, AlignNodes: map[string][]*simpleGraph.Node{"human": []*simpleGraph.Node{humanGraph.Nodes[0]}, "chimp": []*simpleGraph.Node{chimpGraph.Nodes[0]}, "gorilla": []*simpleGraph.Node{gorillaGraph.Nodes[0]}}}
+	nodeAlign1 = graphColumn{AlignId: 1, AlignNodes: map[string][]*simpleGraph.Node{"human": []*simpleGraph.Node{humanGraph.Nodes[1]}}}
+	nodeAlign2 = graphColumn{AlignId: 2, AlignNodes: map[string][]*simpleGraph.Node{"human": []*simpleGraph.Node{humanGraph.Nodes[2]}, "chimp": []*simpleGraph.Node{chimpGraph.Nodes[1]}, "gorilla": []*simpleGraph.Node{gorillaGraph.Nodes[1]}}}
+	nodeAlign3 = graphColumn{AlignId: 3, AlignNodes: map[string][]*simpleGraph.Node{"human": []*simpleGraph.Node{humanGraph.Nodes[3], humanGraph.Nodes[4]}}}
+)
+
+func TestGraphColumn(t *testing.T) {
+	allAlign = append(allAlign, nodeAlign0)
+	allAlign = append(allAlign, nodeAlign1)
+	allAlign = append(allAlign, nodeAlign2)
+	allAlign = append(allAlign, nodeAlign3)
+
+	humanNode1.Next = []*simpleGraph.Edge{humanEdge1, humanEdge2}
+	humanNode2.Next = []*simpleGraph.Edge{humanEdge3}
+	humanNode3.Next = []*simpleGraph.Edge{humanEdge4, humanEdge5}
+	humanNode4.Next = []*simpleGraph.Edge{humanEdge6}
+
+	humanNode2.Prev = []*simpleGraph.Edge{humanEdge1}
+	humanNode3.Prev = []*simpleGraph.Edge{humanEdge2, humanEdge3}
+	humanNode4.Prev = []*simpleGraph.Edge{humanEdge4}
+	humanNode5.Prev = []*simpleGraph.Edge{humanEdge5, humanEdge6}
+
+	chimpNode1.Next = []*simpleGraph.Edge{chimpEdge1}
+	chimpNode2.Prev = []*simpleGraph.Edge{chimpEdge1}
+
+	gorillaNode1.Next = []*simpleGraph.Edge{gorillaEdge1}
+	gorillaNode2.Prev = []*simpleGraph.Edge{gorillaEdge1}
+
+	simpleGraph.PrintGraph(humanGraph)
+	simpleGraph.PrintGraph(chimpGraph)
+	simpleGraph.PrintGraph(gorillaGraph)
+
+	log.Print(nodeAlign0.AlignId)
```

Comment:
Automating this test would be beneficial, even if the check is not perfect.  I worry that this test will always been seen as passing, even if something breaks your code.